### PR TITLE
Fix incomplete kernel search path

### PR DESCRIPTION
**Pull request by Wh1isper**

### CODE REVIEW

**Comment by Wh1isper** [comment]
cc @davidbrochart

**Comment by Wh1isper** [comment]
I missed the change to the kernel driver, will check the ipython kernelspec search path

**Comment by davidbrochart** [comment]
Thanks for the PR @Wh1isper, any idea why tests fail?

**Comment by Wh1isper** [comment]
> Thanks for the PR @Wh1isper, any idea why tests fail?

It seems like starlette switch `requests` to `httpx` and not support `get` with body

https://github.com/encode/starlette/blob/master/starlette/testclient.py#L470

**Comment by Wh1isper** [comment]
refer to this
https://www.python-httpx.org/compatibility/#request-body-on-http-methods
just fixed ci

File: plugins/kernels/fps_kernels/routes.py [hunk 1]
```diff
@@ -57,7 +61,11 @@ async def get_kernelspec(
     file_name,
     user: User = Depends(current_user()),
 ):
-    return FileResponse(prefix_dir / "share" / "jupyter" / "kernels" / kernel_name / file_name)
+    for search_path in kernelspec_dirs():
+        file_path = Path(search_path) / kernel_name / file_name
+        if file_path.exists():
+            return FileResponse(file_path)
+    raise HTTPException(404)
```
Comment (davidbrochart) [inline plugins/kernels/fps_kernels/routes.py:68]:
```suggestion
    raise HTTPException(status_code=404, detail=f"Kernelspec {kernel_name}/{file_name} not found")
```

File: plugins/kernels/fps_kernels/routes.py [hunk 2]
```diff
@@ -127,9 +135,7 @@ async def create_session(
     create_session = CreateSession(**(await request.json()))
     kernel_name = create_session.kernel.name
     kernel_server = KernelServer(
-        kernelspec_path=(
-            prefix_dir / "share" / "jupyter" / "kernels" / kernel_name / "kernel.json"
-        ).as_posix(),
+        kernelspec_path=find_kernelspec(kernel_name),
```
Comment (davidbrochart) [inline plugins/kernels/fps_kernels/routes.py:138]:
```suggestion
        kernelspec_path=Path(find_kernelspec(kernel_name)).as_posix(),
```

File: plugins/kernels/fps_kernels/routes.py [hunk 3]
```diff
@@ -186,12 +192,9 @@ async def execute_cell(
         cell["outputs"] = []
 
         kernel = kernels[kernel_id]
-        kernelspec_path = (
-            prefix_dir / "share" / "jupyter" / "kernels" / kernel["name"] / "kernel.json"
-        ).as_posix()
         if not kernel["driver"]:
             kernel["driver"] = driver = KernelDriver(
-                kernelspec_path=kernelspec_path,
+                kernelspec_path=find_kernelspec(kernel["name"]),
```
Comment (davidbrochart) [inline plugins/kernels/fps_kernels/routes.py:197]:
```suggestion
                kernelspec_path=Path(find_kernelspec(kernel["name"])).as_posix(),
```

**Review by davidbrochart — CHANGES_REQUESTED** [review]

**Comment by davidbrochart** [comment]
> It seems like starlette switch `requests` to `httpx` and not support `get` with body

Should we require `starlette >=0.21.0`?

**Comment by Wh1isper** [comment]
> > It seems like starlette switch `requests` to `httpx` and not support `get` with body
> 
> Should we require `starlette >=0.21.0`?

`fastapi ==0.87.0` will get `starlette==0.21.0`, require `fastapi >=0.87.0` should be ok?

I'm new to `fastapi`, it looks like fastapi will follow up with `starlette` for updates

**Comment by davidbrochart** [comment]
> `fastapi ==0.87.0` will get `starlette==0.21.0`, require `fastapi >=0.87.0` should be ok?

Yes, let's require `fastapi >=0.87.0` then :+1:

**Comment by davidbrochart** [comment]
Just as a side note, you should create a branch on your fork before making a PR, instead of committing on `main`.

**Comment by Wh1isper** [comment]
> Just as a side note, you should create a branch on your fork before making a PR, instead of committing on `main`.

Thanks for the reminder, I realized that each PR will be squeezed into a new commit rather than just merged

**Comment by davidbrochart** [comment]
In this particular case, I squashed your PR.

**Pull request merged**